scipy/ndimage/filters.py

```diff
@@ -274,6 +305,9 @@ def gaussian_filter(input, sigma, order=0, output=None,
     truncate : float
         Truncate the filter at this many standard deviations.
         Default is 4.0.
+    weights : ndarray
```

Comment:
What about making a new `correlatend` function that handles the input/output wapping and the axis iteration? This makes it easier for the user and also avoids adding complexity to the existing functions?